src/common/marlin_client.c

```diff
@@ -144,182 +148,196 @@ void marlin_client_loop(void) {
 }
 
 void marlin_client_set_event_notify(uint64_t notify_events) {
+    const marlin_client_t *client = _client_ptr();
+    if (!client)
+        return;
+
+    const uint8_t cid = client->id;
     char request[MARLIN_MAX_REQUEST];
-    marlin_client_t *client = _client_ptr();
-    if (client) {
-        sprintf(request, "!event_msk %08lx %08lx", (uint32_t)(notify_events & 0xffffffff), (uint32_t)(notify_events >> 32));
-        _send_request_to_server(client->id, request);
-        _wait_ack_from_server(client->id);
-    }
+    sprintf(request, "!event_msk %08lx %08lx", (uint32_t)(notify_events & 0xffffffff), (uint32_t)(notify_events >> 32));
+    _send_request_to_server(cid, request);
+    _wait_ack_from_server(cid);
 }
 
 void marlin_client_set_change_notify(uint64_t notify_changes) {
+    const marlin_client_t *client = _client_ptr();
+    if (!client)
+        return;
+
+    const uint8_t cid = client->id;
     char request[MARLIN_MAX_REQUEST];
-    marlin_client_t *client = _client_ptr();
-    if (client) {
-        sprintf(request, "!change_msk %08lx %08lx", (uint32_t)(notify_changes & 0xffffffff), (uint32_t)(notify_changes >> 32));
-        _send_request_to_server(client->id, request);
-        _wait_ack_from_server(client->id);
-    }
+    sprintf(request, "!change_msk %08lx %08lx", (uint32_t)(notify_changes & 0xffffffff), (uint32_t)(notify_changes >> 32));
+    _send_request_to_server(cid, request);
+    _wait_ack_from_server(cid);
 }
 
 uint32_t marlin_command(void) {
     marlin_client_t *client = _client_ptr();
-    if (client)
-        return client->command;
-    return MARLIN_CMD_NONE;
+    if (!client)
+        return MARLIN_CMD_NONE;
+
+    return client->command;
 }
 
 void marlin_stop_processing(void) {
     marlin_client_t *client = _client_ptr();
-    if (client == 0)
+    if (!client)
         return;
+
     _send_request_to_server(client->id, "!stop");
     _wait_ack_from_server(client->id);
 }
 
 void marlin_start_processing(void) {
     marlin_client_t *client = _client_ptr();
-    if (client == 0)
+    if (!client)
         return;
+
     _send_request_to_server(client->id, "!start");
     _wait_ack_from_server(client->id);
 }
 
 int marlin_motion(void) {
-    marlin_vars_t *vars = marlin_update_vars(MARLIN_VAR_MSK(MARLIN_VAR_MOTION));
-    return (vars->motion) ? 1 : 0;
+    const marlin_vars_t *vars = marlin_update_vars(MARLIN_VAR_MSK(MARLIN_VAR_MOTION));
+    return SIGN(vars->motion);
 }
 
 int marlin_wait_motion(uint32_t timeout) {
-    marlin_vars_t *vars = marlin_update_vars(MARLIN_VAR_MSK(MARLIN_VAR_MOTION));
-    uint32_t tick = HAL_GetTick();
+    const marlin_vars_t *vars = marlin_update_vars(MARLIN_VAR_MSK(MARLIN_VAR_MOTION));
+    const uint32_t tick = HAL_GetTick();
     while ((vars->motion == 0) && ((HAL_GetTick() - tick) < timeout))
         marlin_update_vars(MARLIN_VAR_MSK(MARLIN_VAR_MOTION));
-    return (vars->motion) ? 1 : 0;
+    return SIGN(vars->motion);
 }
```

Comment:
I thing use SIGN is confusing, motion is bitmask, original code is more readable.

Reply:
What about `return (bool) vars->motion`?

Reply:
that's not the same - please keep the original code.